tensorflow/compiler/tf2xla/kernels/reduction_ops.cc

```diff
@@ -84,7 +84,21 @@ REGISTER_XLA_OP(Name("Min").CompileTimeConstantInput("reduction_indices"),
 class MaxOp : public XlaReductionOp {
  public:
   explicit MaxOp(OpKernelConstruction* ctx)
-      : XlaReductionOp(ctx, ctx->input_type(0)) {}
+      : XlaReductionOp(ctx, ctx->input_type(0)) {
+    OP_REQUIRES_OK(ctx, TypeCheck(xla_reduction_type_));
+  }
+
+  Status TypeCheck(xla::PrimitiveType xla_reduction_type_) {
```

Comment:
This should probably be a `static` function because it does not need to access any class members.